### PR TITLE
fix: move proxy normalization

### PR DESCRIPTION
**Pull request by anshumanv**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

existing

### Motivation / Use-Case
move option normalization to normalizer

### Breaking Changes
no

### CODE REVIEW

File: lib/utils/normalizeOptions.js [hunk 1]
```diff
@@ -189,6 +190,72 @@ function normalizeOptions(compiler, options, logger) {
...
+   * Assume a proxy configuration specified as:
+   * proxy: {
+   *   'context': { options }
+   * }
+   * OR
+   * proxy: {
+   *   'context': 'target'
+   * }
+   */
+  if (!Array.isArray(options.proxy)) {
```
[the remark this answers is not in the document]
Comment (anshumanv) [reply lib/utils/normalizeOptions.js]: done

**Review by alexander-akait — APPROVED** [review]
Thanks! I think we need implement the new format of `this.proxy = [{ target:   }]`

**Comment by alexander-akait** [comment]
Maybe you will found other places which can be moved in normize

**Comment by anshumanv** [comment]
> Thanks! I think we need implement the new format of

Will take care 👍 

> Maybe you will found other places which can be moved in normalize

I think none left now, some are there but that's at runtime eg setting port which we can't know beforehand

**Review by snitin315 — APPROVED** [review]

**Comment by snitin315** [comment]
> Thanks! I think we need implement the new format of `this.proxy = [{ target, options, filter }]`

I think we have an open PR for same -> https://github.com/webpack/webpack-dev-server/pull/3224

**Comment by alexander-akait** [comment]
@snitin315 Yep, but with deprecation we should bring new syntax and update deps

**Comment by anshumanv** [comment]
Can someone merge? No access

**Pull request merged**